### ISSUE TITLE
Detection missing

### ISSUE THREAD

**Issue by xiaosuzhang** [AlexeyAB/darknet#1021]
I trained the yolov3 on the celeba face datasets for 20000 iteartion. I select 190000 image form the celeba face datsets  as the training datasets and the rest images as the test datastes. And get the loss figure like this:
![iteration_20000](https://user-images.githubusercontent.com/34598992/41202059-4ad4f19c-6cf6-11e8-9e0f-bcdf0ada962b.jpg)
I got the map from the test datasets is 90%
But when i used the trained-20000-iteration model to detect the images from the FDDB which contain more than one people, it only detect one face. Like this:

![_20180610213336](https://user-images.githubusercontent.com/34598992/41202048-1c1126d2-6cf6-11e8-87de-749fd8620b06.jpg)

I want to know why this happen???
Thanks!!

**Comment by AlexeyAB** [comment]
* What date of your repository?

* Do you get such result for most of images or only for this image?

* Show screenshot of mAP and TP/FP/FN

* What can you get by using this command?
   `./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 416 -height 416`

**Comment by xiaosuzhang** [comment]
I think the calculated anchor is wrong. And i recalculated anchors and get the results:
darknet.exe detector calc_anchors data/face.data -num_of_clusters 9 -width 416 -height 416

 num_of_clusters = 9, width = 416, height = 416
 read labels from 190000 images
 loaded          image: 190000   box: 190000
 all loaded.

 calculating k-means++ ... i = 8441, box_w = 0, box_h = -1610612736, anchor_w = 1073741824, anchor_h = 1610612736, iou = 1.120950
 i = 29536, box_w = 0, box_h = 1610612736, anchor_w = 1073741824, anchor_h = 1610612736, iou = 3.720325
 i = 38214, box_w = 536870912, box_h = 536870912, anchor_w = 1073741824, anchor_h = 1610612736, iou = 1.614132
 i = 64587, box_w = 0, box_h = -1073741824, anchor_w = 1073741824, anchor_h = 1610612736, iou = 3.780674
 i = 119094, box_w = -1073741824, box_h = -1610612736, anchor_w = 1073741824, anchor_h = 1610612736, iou = 17.228031
 i = 134736, box_w = 1610612736, box_h = -1610612736, anchor_w = 1073741824, anchor_h = 1610612736, iou = 2.191319
 i = 162132, box_w = 0, box_h = -1073741824, anchor_w = 1073741824, anchor_h = 1610612736, iou = 3.836589
 i = 162387, box_w = 0, box_h = -536870912, anchor_w = 1073741824, anchor_h = 1610612736, iou = 1.782156
 i = 184663, box_w = -2147483648, box_h = -536870912, anchor_w = 1073741824, anchor_h = 1610612736, iou = 2.576230

 avg IoU = 82.89 %

Saving anchors to the file: anchors.txt
anchors = 81.8529,91.4268, 133.8685,129.8104, 101.6910,184.3668, 176.7015,175.8717, 146.9466,254.7336, 217.6496,219.2072, 199.7912,320.8246, 260.3177,266.3301, 307.4865,331.7714

Is the result is good???

**Comment by AlexeyAB** [comment]
* At least 9 of your labels are wrong

* Looks like these anchors much more suitable for your dataset than default anchors from `yolov3.cfg`

* Much better if you will update your code from GitHub, with recent fixes

**Comment by xiaosuzhang** [comment]
1:
**At least 9 of your labels are wrong??**
what do you mean? Should i recreate the label???
2:
Should i use the recalculated anchors to train my model??? If i use the recalculated anchors to train the model, does it work???
3:
I have updates the code.

**Comment by AlexeyAB** [comment]
>  i = 8441, box_w = 0, box_h = -1610612736, anchor_w = 1073741824, anchor_h = 1610612736, iou = 1.120950

1. These messages says that you have label with negative height = -1610612736

2. Yes, use re-calculated anchors

**Comment by xiaosuzhang** [comment]
Thanks, i will try your solution and train the model. If i have some issues, please give me some suggestions??
